### DISCUSSION TITLE
Solved: "slow stream, dropouts are possible"

### DISCUSSION THREAD

**Discussion by brentfraser** [schreibfaul1/ESP32-audioI2S#528] [General]
I'm running a LilyGo ESP32 T-Display-S3 (ESP32-S3 with a small display) and was experiencing "slow stream, dropouts are possible".

To solve the problem, make sure PSRAM is being used (if your ESP32 has some).   If you are using PlatformIO, in the project's platformio.ini set 

```
build_flags = 
    -DBOARD_HAS_PSRAM
```
For an excellent example, see https://github.com/schreibfaul1/ESP32-audioI2S/blob/7ffc573c51f0321643161702784dbf3b97aa78cc/examples/Better_WiFi_throughput/platformio.ini#L55

This resolved the dropouts for me.  You can confirm the PSRAM availability by adding the following to your code (e.g. in the setup() function):

```
  Serial.println("Total PSRAM: " + String( ESP.getPsramSize() ) );
  Serial.println("Free PSRAM:  " + String( ESP.getFreePsram() ) );
```

To solve the problem, I originally changed from using urls with https to http.  This helped a lot, but did not completely solve it.  Then I was planning to add an external WiFi antenna, but decided to do more reading.  

Hope this helps!

**Comment by schreibfaul1** [comment]
if you use PSRAM, you can still optimize this in menuconfig (for https):
 LWIP ---> Enable IRAM optimization
mbedTLS ---> Memoryallocation starategy (External SPIRAM)

**Comment by brentfraser** [comment]
Thanks for the info.  Just to be clear in case others are reading this, the simple addition of -DBOARD_HAS_PSRAM solved my dropouts problem.

**Comment by cemox** [comment]
When I try to run idf.py menuconfig, I receive two warnings and one error . I really would like to configure with menuconfig  my arduino project (in order to get rid of low dropout warnings) but I don't know how. IDF version is 5.1.12.
```

CMake Warning (dev) in CMakeLists.txt:
  No project() command is present.  The top-level CMakeLists.txt file must
  contain a literal, direct call to the project() command.  Add a line of
  code such as

    project(ProjectName)

  near the top of the file, but after cmake_minimum_required().

  CMake is pretending there is a "project(Project)" command on the first
  line.
This warning is for project developers.  Use -Wno-dev to suppress it.

-- The C compiler identification is GNU 11.4.0
-- The CXX compiler identification is GNU 11.4.0
-- Detecting C compiler ABI info
-- Detecting C compiler ABI info - done
-- Check for working C compiler: /usr/bin/cc - skipped
-- Detecting C compile features
-- Detecting C compile features - done
-- Detecting CXX compiler ABI info
-- Detecting CXX compiler ABI info - done
-- Check for working CXX compiler: /usr/bin/c++ - skipped
-- Detecting CXX compile features
-- Detecting CXX compile features - done
CMake Error at CMakeLists.txt:5 (idf_component_register):
  Unknown CMake command "idf_component_register".


CMake Warning (dev) in CMakeLists.txt:
  No cmake_minimum_required command is present.  A line of code such as

    cmake_minimum_required(VERSION 3.25)

  should be added at the top of the file.  The version specified may be lower
  if you wish to support older CMake versions for this project.  For more
  information run "cmake --help-policy CMP0000".
This warning is for project developers.  Use -Wno-dev to suppress it.

-- Configuring incomplete, errors occurred!
See also "/home/cemo/ESP32-audioI2S/build/CMakeFiles/CMakeOutput.log".
cmake failed with exit code 1



```

**Reply by cniedzi** [reply]
How to solve 'slow stream' in Arduino IDE (PSRAM activated)?

**Reply by cemox** [reply]
Check the betterwifithroughput jpeg in the examples.

**Reply by cniedzi** [reply]
> Check the betterwifithroughput jpeg in the examples.

Could you please share the full path in examples submenu? I couldn't find it :(

**Reply by cemox** [reply]
examples/Better_WiFi_throughput/better_WiFi_throughput.jpeg

**Reply by cniedzi** [reply]
> examples/Better_WiFi_throughput/better_WiFi_throughput.jpeg

Thx. I'm affraid it is not to set in ArduinoIDE...

**Comment by cemox** [comment]
I recommend you to use VScode + platformio.

**Reply by cniedzi** [reply]
It is something totally new for me :( May be I will try to learn it.
